Tribler/Policies/BoostingManager.py

```diff
@@ -0,0 +1,676 @@
+        """
+        if infohash not in self.torrents:
+            return
+
+        self._logger.debug("infohash %s %s %s updated", subject, change_type, hexlify(infohash))
+
+        tdict = self.torrent_db.getTorrent(infohash, keys=['C.torrent_id', 'infohash', 'name',
+                                                           'length', 'category', 'status', 'num_seeders',
+                                                           'num_leechers'])
+
+        if tdict:
+            infohash_str = hexlify(tdict['infohash'])
+
+            new_seed = tdict['num_seeders']
+            new_leecher = tdict['num_leechers']
+
+            if new_seed - self.torrents[tdict['infohash']]['num_seeders'] \
+                    or new_leecher - self.torrents[tdict['infohash']]['num_leechers']:
+                self.torrents[tdict['infohash']]['num_seeders'] = new_seed
+                self.torrents[tdict['infohash']]['num_leechers'] = new_leecher
+                self._logger.info("infohash %s changed s:%d l:%d", infohash_str, new_seed, new_leecher)
+
+    def scrape_trackers(self):
+        """
+        Manually scrape tracker by requesting to tracker manager
+        """
+
+        for infohash, _ in self.torrents.iteritems():
+            # tf_meta = torrent['metainfo']
+
+            # torrent handle
+            # lt_torrent = self.session.lm.ltmgr.get_session().find_torrent(lt.big_number(infohash))
+
+            # check health(seeder/leecher)
+            self.session.lm.torrent_checker.add_gui_request(infohash, True)
+
+        self.session.lm.threadpool.add_task(self.scrape_trackers, self.settings.tracker_interval, "CreditMining_scrape")
+
+    def set_archive(self, source, enable):
+        """
+        setting archive of a particular source. Affect all the torrents in this source
+        """
+        if source in self.boosting_sources:
+            self.boosting_sources[source].archive = enable
+            self._logger.info("Set archive mode for %s to %s", source, enable)
+        else:
+            self._logger.error("Could not set archive mode for unknown source %s", source)
+
+    def start_download(self, torrent):
+        """
+        Start downloading a particular torrent and add it to download list in Tribler
+        """
+        def do_start():
+            """
+            add the actual torrent to the manager to download it later
+            :return:
+            """
+            dscfg = DownloadStartupConfig()
+            dscfg.set_dest_dir(self.settings.credit_mining_path)
+            dscfg.set_safe_seeding(False)
+
+            # just a debug variable
+            tobj = torrent
+
+            preload = tobj.get('preload', False)
+
+            # not using Session.start_download because we need to specify pstate
+            try:
+                assert self.session.get_libtorrent()
+                assert not self.session.lm.download_exists(tobj["metainfo"].get_infohash())
```

Comment:
Using assertions to verify a condition is not very nice. Why not use an if-statement instead?